specification/data-resource-semantic-conventions.md

```diff
@@ -70,6 +70,28 @@ Labels defining a compute unit (e.g. Container, Process, Lambda Function).
 | container.image.name | Name of the image the container was built on. | `gcr.io/opentelemetry/operator` |
 | container.image.tag | Container image tag. | `0.1` |
 
+
+### Function as a Service
+
+**type:** `faas`
+
+**Description:** A serverless instance.
+
+| Label  | Description  | Example  | Required |
+|---|---|---|--|
+| faas.region| A specific geographical location where the function is executed. | `us-east-2` | Yes |
+| faas.name | The name of the function being executed. | `opentelemetry-lambda` | Yes |
+| faas.identifier | The unique name of the function being executed. <br /> For example, in AWS this field correspond to the [ARN] value, in GCP to the URI of the resource, and Azure to the [FunctionDirectory] field. | `opentelemetry-lambda` | Yes |
+| faas.version | The version of the function being executed. | `opentelemetry-latest` | No |
+| faas.provider | Provider name of the function, e.g. `aws`, `gpc`, `azure`, `serverless`, ... | `aws` | No |
+| faas.payload | The input passed to the function. | `{name: 'opentelemetry'}` | No |
+| faas.trigger | Type of the trigger that spawned the function, e.g. `http`, `S3`, `PubSub`, ... | `http` | Yes |
```

Comment:
`S3` is a service, but `http` is a transport. If you want this to be the transport used, then I would remove the `S3` trademark name and use all generic transports like http and PubSub. If you want this to be the service that triggered the function, then I would use examples that are all service names like API Gateway, S3, SQS.

Reply:
Makes sense! Maybe an enum to chose from is better. Say: http, pubsub, event, timer, manual

Reply:
The function may not know what was the actual trigger. This could be completely abstracted by the FaaS platform (e.g. harmonize http/https and http2).

Reply:
Let's make this optional, then.

Reply:
I would keep it mandatory. It is true that the function might not know the exact protocol used (e.g. http or https), but conceptually the function has the knowledge if it answers to an HTTP request, to a PubSub event, to a cron job, and so forth.

Reply:
Fair enough, if we are confident that this information is available to the instrumentation code, let's keep it mandatory.